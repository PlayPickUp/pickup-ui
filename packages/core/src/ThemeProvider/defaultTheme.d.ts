import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
