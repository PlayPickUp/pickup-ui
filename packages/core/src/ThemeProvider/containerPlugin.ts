/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
