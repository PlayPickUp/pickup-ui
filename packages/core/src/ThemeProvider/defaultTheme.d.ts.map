{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}