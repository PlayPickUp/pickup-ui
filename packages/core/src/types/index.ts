import { FieldInputProps, FormikErrors, FormikTouched } from "formik";
import React, {
  ChangeEvent,
  CSSProperties,
  MouseEvent,
  ReactElement,
  ReactNode,
  SetStateAction,
  Dispatch,
} from "react";

/**
 *
 * Theme Specific Types
 *
 */
export interface ThemeProviderProps {
  children: any;
  withReset?: boolean;
  theme?: DefaultTheme;
}

export interface PurpleMap {
  light: string;
  base: string;
  dark: string;
}

export interface GreenMap {
  light: string;
  base: string;
}

export interface GreyMap {
  light: string;
  base: string;
  dark: string;
}

export interface RedMap {
  light: string;
  base: string;
}

export interface DefaultThemeColors {
  primary: PurpleMap;
  secondary: GreyMap;
  green: GreenMap;
  purple: PurpleMap;
  black: string;
  white: string;
  success: string;
  grey: GreyMap;
  red: RedMap;
  error: string;
}

export interface FontFamilies {
  body: string;
  headline: string;
}

interface FontProperties {
  fontFamily: string;
  fontWeight: number | string;
  fontStyle?: string;
  fontSize: number | string;
  lineHeight: string;
  letterSpacing: string;
  textTransform?: string;
  color: string;
}

export interface FontStyles {
  title: FontProperties;
  heading2: FontProperties;
  heading3: FontProperties;
  body: FontProperties;
  body2: FontProperties;
}

export interface ResponsiveFonts {
  desktop: FontStyles;
  mobile: FontStyles;
}

export interface DefaultTypography {
  fontFamilies: FontFamilies;
  fontStyles: ResponsiveFonts;
}
export type CreateShadow = (
  a: number,
  b: number,
  c: number,
  d: number,
  e: number,
  f: number,
  g: number,
  h: number,
  i: number,
  j: number,
  k: number,
  l: number
) => string;

export type DefaultSpacing = {
  base: number;
};

export type DefaultShadows = Array<string>;

export interface DefaultZIndex {
  overlay: number;
}

export interface DefaultGradients {
  main: string;
  hover: string;
}

export type MediaQuery = (minWidth: number) => string;

export interface Breakpoints {
  small: number;
  medium: number;
  large: number;
}

export type DefaultTheme = {
  borderRadius: number;
  breakpoints: Breakpoints;
  mediaQuery: MediaQuery;
  colors: DefaultThemeColors;
  gradients: DefaultGradients;
  shadows: DefaultShadows;
  spacing: DefaultSpacing;
  typography: DefaultTypography;
  zIndex: DefaultZIndex;
};

/**
 *
 * Button Component
 *
 */

export type ButtonVariant = "fit" | "full";

export interface ButtonProps {
  children?: ReactNode;
  disabled?: boolean;
  style?: React.CSSProperties;
  variant?: ButtonVariant;
  onClick?: (e?: React.MouseEvent) => unknown;
  element?: keyof JSX.IntrinsicElements;
  to?: string;
  href?: string;
  className?: string;
  useSubmit?: boolean;
  submitText?: string;
  color?: "primary" | "secondary" | "light";
}

/**
 *
 * Typography Component
 *
 */

export interface TypographyProps {
  children: ReactNode;
  variant?:
    | "title"
    | "body"
    | "body2"
    | "heading2"
    | "heading3"
    | "heading4"
    | "heading5"
    | "heading6"
    | "span";
  className?: string;
  element?: "p" | "span" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  style?: CSSProperties;
  color?: string;
  useUnescape?: boolean;
}

export type TypographyElementMap = Record<string, keyof JSX.IntrinsicElements>;

/**
 *
 * Fab (Floating Action Button) Component
 *
 */

export interface FabProps {
  className?: string;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  style?: React.CSSProperties;
  icon: any;
  title: string;
  size?: "small" | "large";
  onClick: (e: MouseEvent) => unknown;
  disabled?: boolean;
  disablePopOver?: boolean;
  color?: string;
  iconColor?: string;
}

/**
 *
 * Table Component
 *
 */
export interface HeadCell {
  disablePadding: boolean;
  id: keyof TableData;
  label: string;
  numeric?: boolean;
  width?: number | undefined;
}

export type TableData = Record<any, any>;

export type TableOrder = "asc" | "desc";
export interface DataTableProps {
  headCells: HeadCell[];
  rows: Array<Record<string | number, any>>;
  defaultSortColumn: string;
  tableTitle: string;
  actionToolbar: (props: any) => ReactElement;
}

export interface EnhancedTableProps {
  classes: ReturnType<any>;
  numSelected: number;
  onRequestSort: (event: React.MouseEvent<unknown>, property: any) => void;
  onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>) => void;
  order: TableOrder;
  orderBy: string;
  rowCount: number;
  headCells: HeadCell[];
}

export interface EnhancedTableToolbarProps {
  numSelected: number;
  tableTitle: string;
  actionToolbar: (props: any) => ReactElement;
  selected: string[];
  setSelected: Dispatch<SetStateAction<string[]>>;
}

/**
 *
 * Icon Component
 *
 */
export interface IconBaseProps {
  children: ReactElement;
  color?: string;
  size?: number;
  className?: string;
  style?: React.CSSProperties;
}

export interface BaseFormikFields {
  id: string;
  name: string;
  field: FieldInputProps<any>;
  form: Record<any, any>;
}

export type BaseFormikHandleChange = (e: ChangeEvent) => unknown;

export interface BaseFormikFieldsWithHandleChange extends BaseFormikFields {
  handleChange: BaseFormikHandleChange;
}

/**
 *
 * TextInput Component
 *
 */
export interface TextInputProps extends BaseFormikFieldsWithHandleChange {
  className?: string;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  style?: React.CSSProperties;
  placeholder?: string;
  label?: string;
}

/**
 *
 * Label Component
 *
 */
export interface LabelProps {
  htmlFor: string;
  children: string | ReactNode;
  className?: string;
  style?: React.CSSProperties;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
}

/**
 *
 * TextArea Component
 *
 */
export interface TextAreaProps extends BaseFormikFieldsWithHandleChange {
  className?: string;
  style?: React.CSSProperties;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  label?: string;
}

/**
 *
 * Select Component
 *
 */ export interface SelectItem {
  label: string;
  value: string;
}

export type BaseFormikSetFieldValue = (
  field: string,
  value: any,
  shouldValidate?: boolean | undefined
) => unknown;

export interface SelectProps extends BaseFormikFields {
  className?: string;
  items: SelectItem[];
  label: string;
  placeholder?: string;
  style?: React.CSSProperties;
}

export interface DropdownStyleProps {
  isOpen: boolean;
}

/**
 *
 * NestedInput Component
 *
 */
export interface NestedInputProps extends BaseFormikFieldsWithHandleChange {
  buttonText: string;
  placeholder: string;
}

/**
 *
 * MultiSelect Component
 *
 */
export interface MultiSelectProps extends BaseFormikFields {
  items: SelectItem[];
  label: string;
}

/**
 *
 * Individual Icon Component
 *
 */
export interface IconProps {
  color?: string;
  className?: string;
  style?: React.CSSProperties;
}

/**
 *
 * Chip Component
 *
 */
export interface ChipProps {
  disabled?: boolean;
  onClick?: (e: React.MouseEvent) => unknown;
  color?: "default" | "primary" | "secondary";
  className?: string;
  style?: React.CSSProperties;
  isActive?: boolean;
  label: string;
  element?: keyof JSX.IntrinsicElements;
  href?: string;
}

/**
 *
 * Paper Component
 *
 */
export interface PaperProps {
  padding?: number;
  children: unknown;
  style?: React.CSSProperties;
  className?: string;
}

/**
 *
 * FormError Component
 *
 */
export interface FormErrorProps {
  errors: FormikErrors<unknown>;
  touched: FormikTouched<unknown>;
  name: string;
}

/**
 *
 * Loader Component
 *
 */
export interface LoaderProps {
  className?: string;
  style?: React.CSSProperties;
  delay?: number;
  text?: string;
}

/**
 *
 * Picker Button
 *
 */
export interface PickerButtonProps {
  className?: string;
  displayText: string;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  style?: React.CSSProperties;
  showResult?: boolean;
  result: number | null;
  isPick: boolean;
}
