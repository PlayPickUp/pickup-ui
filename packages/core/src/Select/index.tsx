import React, { useState } from "react";
import { createUseStyles, useTheme } from "react-jss";
import classNames from "classnames";
import Downshift from "downshift";
import Label from "../Label";
import Icon from "../Icon";
import ChevronDown from "../Icon/icons/ChevronDown";

import { DefaultTheme, SelectItem, SelectProps } from "../types";
import FormError from "../FormError";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    position: "relative",
    width: "100%",
  },
  selectButton: {
    position: "relative",
    height: 44,
    width: "100%",
    fontFamily: theme.typography.fontFamilies.body,
    fontSize: 15,
    lineHeight: "18px",
    letterSpacing: "0.1px",
    color: theme.colors.black,
    backgroundColor: theme.colors.white,
    border: `1px solid ${theme.colors.grey.base}`,
    borderRadius: theme.borderRadius,
    padding: `0 ${theme.spacing.base * 5}px 0 ${theme.spacing.base * 4}px`,
    textAlign: "left",
    "&:focus, &:active": {
      border: `1px solid ${theme.colors.grey.base}`,
      outline: "none",
    },
  },
  iconButton: {
    position: "absolute",
    right: `${theme.spacing.base * 3}px`,
    top: `${theme.spacing.base * 8.75}px`,
    "&:active, &:focus": {
      outline: "none",
    },
  },
  icon: {
    position: "relative",
    transition: "all 250ms ease-out",
    transform: "rotate(0deg)",
  },
  iconOpen: {
    transform: "rotate(180deg)",
  },
  dropdown: {
    position: "relative",
    top: -3,
    height: "auto",
    width: "100%",
    fontFamily: theme.typography.fontFamilies.body,
    fontSize: 14,
    lineHeight: "18px",
    letterSpacing: "0.1px",
    color: theme.colors.black,
    backgroundColor: theme.colors.white,
    border: `1px solid ${theme.colors.grey.base}`,
    borderTop: "transparent",
    borderRadius: theme.borderRadius,
    borderTopLeftRadius: 0,
    borderTopRightRadius: 0,
    textAlign: "left",
    "&:focus, &:active": {
      border: `1px solid ${theme.colors.grey.base}`,
      borderTop: "transparent",
      outline: "none",
    },
  },
  dropdownItem: {
    padding: `${theme.spacing.base * 4}px`,
  },
}));

const Select: React.FC<SelectProps> = ({
  className,
  items,
  label,
  placeholder = "Select an item...",
  style,
  ...props
}) => {
  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);

  const theme = useTheme<DefaultTheme>();
  const classes = useStyles();

  const handleSelection = (name: string, selection: SelectItem): void => {
    if (!selection) {
      return setIsMenuOpen(false);
    }
    props.form.setFieldValue(name, selection.value);
    setIsMenuOpen(false);
  };

  return (
    <>
      <Downshift
        isOpen={isMenuOpen}
        onChange={(selection) => handleSelection(props.field.name, selection)}
        itemToString={(item: SelectItem) => (item ? item.label : "")}
        onOuterClick={() => setIsMenuOpen(false)}
      >
        {({
          getInputProps,
          getItemProps,
          getLabelProps,
          getMenuProps,
          getToggleButtonProps,
          isOpen,
          inputValue,
          highlightedIndex,
          selectedItem,
        }) => {
          return (
            <div className={classNames(classes.root, className)} style={style}>
              <Label {...getLabelProps()}>{label}</Label>
              <input
                className={classes.selectButton}
                {...getInputProps({
                  placeholder,
                  ...props.field,
                })}
              />
              <button
                {...getToggleButtonProps({
                  className: classes.iconButton,
                  onClick: () => setIsMenuOpen(!isMenuOpen),
                })}
                aria-label="toggle-menu"
              >
                <Icon
                  className={classNames({
                    [classes.icon]: true,
                    [classes.iconOpen]: isOpen,
                  })}
                  color={theme.colors.grey.base}
                >
                  <ChevronDown />
                </Icon>
              </button>
              <ul
                className={classes.dropdown}
                {...getMenuProps()}
                style={{ display: isOpen ? "block" : "none" }}
              >
                {isOpen
                  ? items
                      .filter(
                        (item) =>
                          !inputValue ||
                          item.label
                            .toLowerCase()
                            .includes(inputValue.toLowerCase())
                      )
                      .map((item, index) => (
                        <li
                          key={`${item.value}${index}`}
                          className={classes.dropdownItem}
                          {...getItemProps({
                            // key: item.value,
                            index,
                            item,
                            style: {
                              backgroundColor:
                                highlightedIndex === index
                                  ? theme.colors.purple.light
                                  : theme.colors.white,
                              fontWeight:
                                selectedItem === item ? "bold" : "normal",
                            },
                          })}
                        >
                          {item.label}
                        </li>
                      ))
                  : null}
              </ul>
            </div>
          );
        }}
      </Downshift>
      <FormError
        errors={props.form.errors}
        touched={props.form.touched}
        name={props.field.name}
      />
    </>
  );
};

export default Select;
