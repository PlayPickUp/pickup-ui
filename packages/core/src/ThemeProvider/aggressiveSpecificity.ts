/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
