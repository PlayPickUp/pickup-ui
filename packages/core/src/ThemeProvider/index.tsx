import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
