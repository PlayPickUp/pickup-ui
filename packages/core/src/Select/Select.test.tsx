/**
 *
 * NOTE: These tests will fail in MultiDownshift without the plugin
 * '@babel/plugin-proposal-class-properties in the babel.config file
 *
 **/

import React from "react";
import jssSerializer from 'jss-snapshot-serializer';
expect.addSnapshotSerializer(jssSerializer);
import { render, fireEvent, cleanup, waitFor } from "@testing-library/react";
import { Formik, Field, Form } from "formik";
import ThemeProvider from "../ThemeProvider";
import Select from ".";
import { SelectItem } from "../types";

const sleep = (ms) => new Promise((r) => setTimeout(r, ms));

// Props
const items: SelectItem[] = [
  {
    label: "Option 1",
    value: "option1",
  },
  {
    label: "Option 2",
    value: "option2",
  },
  {
    label: "Option 3",
    value: "option3",
  },
];

const FormComponent: React.FC<{
  onSubmit: jest.Mock;
}> = ({ onSubmit }) => {
  const handleSubmit = async (values) => {
    await sleep(500);
    onSubmit(values);
  };
  return (
    <ThemeProvider>
      <div>
        <Formik
          initialValues={{
            selection: "",
          }}
          onSubmit={handleSubmit}
        >
          <Form>
            <Field
              id="selection"
              name="selection"
              label="Select Test"
              placeholder="Select one"
              className="Input-1-2-3"
              style={{ color: "blue" }}
              items={items}
              component={Select}
            />
            <button type="submit">Submit</button>
          </Form>
        </Formik>
      </div>
    </ThemeProvider>
  );
};

const handleSubmit = jest.fn();

test("Select renders without crashing, matches snapshot", () => {
  Math.random = jest.fn(() => 1);  // <--- This is the key, overriding the system's Math.random function
  const { container } = render(<FormComponent onSubmit={handleSubmit} />);
  expect(container).toMatchSnapshot();
});

test("Props are passed and rendered", () => {
  const { getByText, getByRole, getByPlaceholderText, getByTestId } = render(
    <FormComponent onSubmit={handleSubmit} />
  );

  expect(getByText("Select Test")).toBeTruthy();
  expect(getByPlaceholderText("Select one")).toBeTruthy();
  expect(getByTestId("select").getAttribute("class")).toContain("Input-1-2-3");
  expect(getByTestId("select").getAttribute("style")).toEqual("color: blue;");
  const menuButton = getByRole("button", { name: "toggle-menu" });
  fireEvent.click(menuButton);
  expect(getByText("Option 2")).toBeTruthy();
  cleanup;
});

test("Item is selectable", () => {
  const { getByText, getByRole } = render(
    <FormComponent onSubmit={handleSubmit} />
  );

  const menuButton = getByRole("button", { name: "toggle-menu" });
  fireEvent.click(menuButton);
  fireEvent.click(getByText("Option 1"));

  const input = getByRole("textbox", {
    name: "Select Test",
  }) as HTMLInputElement;
  expect(input.value).toEqual("option1");
  cleanup;
});

test("Values are passed to form submission", async () => {
  const { getByText, getByRole } = render(
    <FormComponent onSubmit={handleSubmit} />
  );

  const menuButton = getByRole("button", { name: "toggle-menu" });
  fireEvent.click(menuButton);
  fireEvent.click(getByText("Option 1"));

  fireEvent.click(getByRole("button", { name: /submit/i }));

  await waitFor(() =>
    expect(handleSubmit).toHaveBeenCalledWith({
      selection: "option1",
    })
  );
});
