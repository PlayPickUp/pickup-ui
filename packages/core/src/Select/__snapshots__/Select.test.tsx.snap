// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j1 {
  display: block;
  padding: 4px 4px 12px;
  position: relative;
  font-size: 16px;
  font-family: 'Ringside-Condensed', Helvetica, sans-serif;
  text-transform: uppercase;
}

body #PickUpUI .j4 {
  height: auto;
  position: relative;
}

body #PickUpUI .j5 {
  width: 20px;
}

body #PickUpUI .j0 {
  width: 100%;
  position: relative;
}

body #PickUpUI .j2 {
  color: #2C2933;
  width: 100%;
  border: 1px solid #9D9D9E;
  height: 44px;
  padding: 0 20px 0 16px;
  position: relative;
  font-size: 15px;
  box-sizing: border-box;
  text-align: left;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 18px;
  border-radius: 4px;
  letter-spacing: 0.1px;
  background-color: #FFF;
}

body #PickUpUI .j3 {
  top: 49px;
  color: inherit;
  right: 12px;
  border: none;
  cursor: pointer;
  margin: 0;
  display: inline-block;
  padding: 0;
  position: absolute;
  font-size: inherit;
  font-family: inherit;
  user-select: none;
  border-color: transparent;
  -moz-appearance: none;
  text-decoration: none;
  background-color: transparent;
  -webkit-appearance: none;
}

body #PickUpUI .j6 {
  position: relative;
  transform: rotate(0deg);
  transition: all 250ms ease-out;
}

body #PickUpUI .j7 {
  top: -3px;
  color: #2C2933;
  width: 100%;
  border: 1px solid #9D9D9E;
  height: auto;
  position: relative;
  font-size: 14px;
  border-top: transparent;
  text-align: left;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 18px;
  border-radius: 4px;
  letter-spacing: 0.1px;
  background-color: #FFF;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

<div>
  <div
    id="PickUpUI"
  >
    <div>
      <form
        action="#"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="downshift-0-label"
          class="j0 Input-1-2-3"
          data-testid="select"
          role="combobox"
          style="color: blue;"
        >
          <div
            data-testid="label-wrapper"
          >
            <label
              class="j1"
              data-testid="label"
              for="downshift-0-input"
            >
              Select Test
            </label>
          </div>
          <input
            aria-autocomplete="list"
            aria-labelledby="downshift-0-label"
            autocomplete="off"
            class="j2"
            id="downshift-0-input"
            name="selection"
            placeholder="Select one"
            value=""
          />
          <button
            aria-haspopup="true"
            aria-label="toggle-menu"
            class="j3"
            data-toggle="true"
            role="button"
            type="button"
          >
            <div
              class="j4 j5 j6"
              data-testid="icon"
            >
              <svg
                height="140"
                version="1.1"
                viewBox="0 0 140 140"
                width="140"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  transform="matrix(5.833333333333333,0,0,5.833333333333333,0,0)"
                >
                  <path
                    d="M23.25,7.311,12.53,18.03a.749.749,0,0,1-1.06,0L.75,7.311"
                    fill="none"
                    stroke="#9D9D9E"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                  />
                </g>
              </svg>
            </div>
          </button>
          <ul
            aria-labelledby="downshift-0-label"
            class="j7"
            id="downshift-0-menu"
            role="listbox"
            style="display: none;"
          />
        </div>
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
